### PR TITLE
Added fix for serializing PIL images

### LINKED ISSUE #1522: Support list of Plotly objects
Currently the following code doesn't work:

```python
import reflex as rx
import plotly.express as px
import plotly.graph_objects as go
import pandas as pd


xx = [1,2,3,4]
yy = [3,2,6,1]
df = pd.DataFrame(zip(xx,yy))
fig = px.line(df,x=0,y=1)
class State(rx.State):

    stateFig: go.Figure = fig
    figList: list[go.Figure] = [fig]*4


def index():
    return rx.vstack(
        rx.heading("hello"),
        rx.plotly(data=State.stateFig),
        rx.plotly(data=State.figList[0],height="400px")
    )

app = rx.App(state=State)
app.add_page(index)

app.compile()
```

The first figure is shown, but when indexing into a list, the output is not rendered correctly. We should support this syntax.

### LINKED ISSUE #1578: Cannot serialize a list of PIL.Images, but a single PIL.Image is fine
**Describe the bug**
When trying to store or compute a list of PIL.Image.Image objects an error is thrown:
```
Task exception was never retrieved
future: <Task finished name='Task-281' coro=<AsyncServer._handle_event_internal() done, defined at /Users/craig/miniforge3/envs/decimer/lib/python3.10/site-packages/socketio/asyncio_server.py:522> exception=TypeError("'Image' object is not iterable")>
Traceback (most recent call last):
  File "/Users/craig/miniforge3/envs/decimer/lib/python3.10/site-packages/socketio/asyncio_server.py", line 524, in _handle_event_internal
    r = await server._trigger_event(data[0], namespace, sid, *data[1:])
  File "/Users/craig/miniforge3/envs/decimer/lib/python3.10/site-packages/socketio/asyncio_server.py", line 569, in _trigger_event
    return await self.namespace_handlers[namespace].trigger_event(
  File "/Users/craig/miniforge3/envs/decimer/lib/python3.10/site-packages/socketio/asyncio_namespace.py", line 37, in trigger_event
    ret = await handler(*args)
  File "/Users/craig/miniforge3/envs/decimer/lib/python3.10/site-packages/reflex/app.py", line 737, in on_event
    self.emit(str(constants.SocketEvent.EVENT), update.json(), to=sid)
  File "/Users/craig/miniforge3/envs/decimer/lib/python3.10/site-packages/reflex/base.py", line 33, in json
    return self.__config__.json_dumps(self.dict(), default=list)
  File "/Users/craig/miniforge3/envs/decimer/lib/python3.10/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "/Users/craig/miniforge3/envs/decimer/lib/python3.10/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/Users/craig/miniforge3/envs/decimer/lib/python3.10/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
TypeError: 'Image' object is not iterable
```

**To Reproduce**
Try to save or compute a list of PIL.Image.Image objects to state.

**Expected behavior**
Since reflex can successfully serialize a single PIL.Image.Image from State, I don't see why we couldn't serialize an array.

**Specifics (please complete the following information):**
 - Python Version: 3.10.12
 - Reflex Version: 0.2.4
 - OS:  Darwin (M1)
 - Browser (Optional): Chrome

### PR DESCRIPTION
**Pull request by wassafshahzad**
### All Submissions:

- [x] Have you followed the guidelines stated in [CONTRIBUTING.md](https://github.com/pynecone-io/pynecone/blob/main/CONTRIBUTING.md) file?
- [x] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/pynecone-io/pynecone/pulls ) for the desired changed?

### Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

#### Description
Updates states format_state.py to handle list, set and tuple

#### Linked Issue
Closes #1578
Closes #1522

### CODE REVIEW

File: reflex/utils/format.py [hunk 1]
```diff
@@ -381,7 +381,7 @@ def format_image_data(value: Type) -> str:
     return f"data:image/png;base64,{base64_image}"
 
 
-def format_state(value: Any) -> Dict:
+def format_state(value: Any) -> Any:
```
Comment (wassafshahzad) [inline reflex/utils/format.py:384]: I updated the type here since if a string is passed into format
-state.py then the return Type is also string.

**Comment by masenf** [comment]
## sample code

```python
from PIL import Image, ImageDraw
import reflex as rx

width, height = 200, 200


class State(rx.State):
    """The app state."""
    images: list[Image.Image] = []

    def generate_solid_color(self):
        self.images.append(Image.new("RGB", (width, height), "blue"))

    def generate_checkerboard(self):
        image_checkerboard = Image.new("RGB", (width, height), "white")
        draw = ImageDraw.Draw(image_checkerboard)
        for x in range(0, width, 50):
            for y in range(0, height, 50):
                if (x // 50 + y // 50) % 2 == 0:
                    draw.rectangle([x, y, x + 50, y + 50], fill="black")
        self.images.append(image_checkerboard)

    def generate_gradient(self):
        # Create a red and green gradient pattern
        image_gradients = Image.new("RGB", (width, height))
        draw = ImageDraw.Draw(image_gradients)
        for x in range(width):
            for y in range(height):
                r = x / width * 255
                g = y / height * 255
                draw.point((x, y), (int(r), int(g), 0))
        self.images.append(image_gradients)

    def generate_stripes(self):
        # Create a blue and yellow diagonal stripes pattern
        image_stripes = Image.new("RGB", (width, height), "white")
        draw = ImageDraw.Draw(image_stripes)
        stripe_width = 20
        for i in range(-(width // stripe_width), height // stripe_width):
            y = i * stripe_width
            draw.rectangle([(0, y), (width, y + stripe_width)], fill=("blue" if i % 2 == 0 else "yellow"))
        self.images.append(image_stripes)

    @rx.var
    def reversed_images(self) -> list[Image.Image]:
        return self.images[::-1]


def index() -> rx.Component:
    return rx.fragment(
        rx.connection_modal(),  # displays error when backend cannot connect
        rx.color_mode_button(rx.color_mode_icon(), float="right"),
        rx.vstack(
            rx.heading("PIL Image Generators", font_size="2em"),
            rx.hstack(
                rx.button("Solid Color", on_click=State.generate_solid_color),
                rx.button("Checkerboard", on_click=State.generate_checkerboard),
                rx.button("Gradient", on_click=State.generate_gradient),
                rx.button("Stripes", on_click=State.generate_stripes),
            ),
            rx.responsive_grid(
                rx.foreach(
                    State.reversed_images,
                    lambda i: rx.grid_item(rx.image(src=i)),
                ),
                columns=[3],
                spacing="4",
            ),
            spacing="1.5em",
            padding_top="10%",
        ),
    )


# Add state and page to the app.
app = rx.App()
app.add_page(index)
app.compile()
```

**Review by masenf — APPROVED** [review]
Very nice. Thanks for your contribution!

**Comment by masenf** [comment]
We're preparing for a release Monday, so I'll have to check with the team to see if this will make 0.2.5 (next) or 0.2.6 (next next) release.

**Pull request merged**